### PR TITLE
batch deleteAll

### PR DESCRIPTION
**Pull request by randallalexander**
Proposal for batch deleteAll
  - resolves item #27

### CODE REVIEW

**Comment by philwills** [comment]
Thanks for this.  I'll try to have a proper look later, but first impressions look good.

File: src/main/scala/com/gu/scanamo/ScanamoFree.scala [hunk 1]
```diff
@@ -14,18 +14,34 @@ object ScanamoFree {
...
-    items.grouped(25).toList.traverseU(batch =>
+    items.grouped(batchSize).toList.traverseU(batch =>
       ScanamoOps.batchWrite(
         new BatchWriteItemRequest().withRequestItems(Map(tableName -> batch.toList.map(i =>
           new WriteRequest().withPutRequest(new PutRequest().withItem(f.write(i).getM))
         ).asJava).asJava)
       )
     )
 
+  def deleteAll[T](tableName: String)(deleteItems: (Symbol, Set[T]))(implicit f: DynamoFormat[T]): ScanamoOps[List[BatchWriteItemResult]] = {
```
Comment (philwills) [inline src/main/scala/com/gu/scanamo/ScanamoFree.scala]: I think it would be a bit more consistent for this to take a `Set[UniqueKey[_]]`.

Comment (randallalexander) [inline src/main/scala/com/gu/scanamo/ScanamoFree.scala]: I was trying `Set[UniqueKey[_]]` and ran across `UniqueKeys[_]` which I think is a better fit.  about to push an update.  Let me know what you think.

**Comment by randallalexander** [comment]
I believe I have addressed everything.  Not sure if you wanted me to update the main README.md too.  Assuming "no" but you know what they say about assumptions.

**Comment by philwills** [comment]
Wonderful, thanks @randallalexander!

**Comment by philwills** [comment]
This is now released in `0.8.2`, which should close #27.

I'm currently working on overhauling the documentation, so haven't added anything to the README yet, but didn't think it was necessary to tie that to the release.

**Pull request merged**